pebbles/static/js/controllers/StatsController.js

```diff
@@ -2,9 +2,277 @@ app.controller('StatsController', ['$q', '$scope', '$http', '$interval', '$uibMo
+                var row = '';
+                CSV = headerString;
+                for (var d in jsonp[0].data) {
+                    row = jsonp[0].data[d]['user'] + ',' + jsonp[0].data[d]['institution'] 
+                    + '\r\n';
+                    CSV += row;                   
+                };
+                $scope.downloadCSV(CSV, 'active_users');
+                
+            });
+        };
+
+        $scope.getInstitutionCSV = function() {
+            
+            exportInstitutions.getList($scope.dates).then(function(response){
+                var jsonStr = JSON.stringify(response);
+                var jsonp = JSON.parse(jsonStr);
+                var headerString = 'Institution,Total users\r\n';
+                var CSV = '';
+                var row = '';
+                CSV = headerString;
+                for (var d in jsonp[0].data) {
+                    row = jsonp[0].data[d] + '\r\n';
+                    CSV += row;                   
+                };
+                $scope.downloadCSV(CSV, 'institutions');
+                
+            });
+        };
+
+        $scope.getMonthlyInstancesCSV = function() {
+            exportMonthlyInstances.getList($scope.dates).then(function(response) {
+                var jsonStr = JSON.stringify(response);
+                var jsonp = JSON.parse(jsonStr);
+                var headerString = 'Year,Jan,Feb,Mar,Apr,May,Jun,Jul,Aug,Sep,Oct,Nov,Dec,Total\r\n';
+                var CSV = '';
+                var row = '';
+                CSV = headerString;
+                for (var d in jsonp[0].data) {
+                    row = d + ',' + jsonp[0].data[d] + '\r\n';
+                    CSV += row;
+                };
+                $scope.downloadCSV(CSV, 'monthly_instances');
+            });
+        };
+
+        $scope.getQuartalsCSV = function() {
+            exportQuartals.getList($scope.dates).then(function(response) {
+                var jsonStr = JSON.stringify(response);
+                var jsonp = JSON.parse(jsonStr);
+                var headerString = 'User type,Year,Q1,Q2,Q3,Q4\r\n';
+                var CSV = '';
+                var row = '';
+                CSV = headerString;
+                for (var d in jsonp[0].data) {
+                    row = jsonp[0].data[d]['org'] + ',' + jsonp[0].data[d]['year'] + ','
+                    + jsonp[0].data[d]['quartals'] + '\r\n';
+                    CSV += row;
+                };
+                $scope.downloadCSV(CSV, 'quartals');
+            });
+        };
+
+        $scope.getOrganizationQuartalsCSV = function () {
+            exportQuartalsByOrganization.getList($scope.dates).then(function(response){
+                var jsonStr = JSON.stringify(response);
+                var jsonp = JSON.parse(jsonStr);
+                var headerString = 'Organization,Year,Q1,Q2,Q3,Q4\r\n';
+                var CSV = '';
+                var row = '';
+                CSV = headerString;
```

Comment:
in these functions too - same set of operations done in more than one function. Redundant code can be avoided.